airflow/jobs.py

```diff
@@ -1983,6 +1983,13 @@ def _execute(self):
             )
             self.process = subprocess.Popen(['bash', '-c', command])
             self.logger.info("Subprocess PID is {}".format(self.process.pid))
+            ti = self.task_instance
```

Comment:
@saguziel @aoen you cannot depend on task_instance being fully populated. SQL Merging here made "try_number" always be 0, so retries will try forever. I have a PR out for that in a few minutes and hopefully a test with it. But I wonder what is happening in your production?